webgl build failed

**Describe the bug**
I install emsdk as the building manual told me and build desktop release, but there is some error when i run `./build.sh -p webgl release` command. i can't build web successfully.

**To Reproduce**
Steps to reproduce the behavior:
1. install emsdk 3.1.15
2. run `build.sh -i -p desktop release`
3. run `build .sh -p webgl release`


**Logs**
```
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:152:16: error: redefinition of 'RegisterClassMethod<ReturnType (ClassType::*)(Args...) noexcept>'
        struct RegisterClassMethod<ReturnType (ClassType::*)(Args...) noexcept> {
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1369:8: note: previous definition is here
struct RegisterClassMethod<ReturnType (ClassType::*)(Args...) noexcept>
       ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:175:16: error: redefinition of 'RegisterClassMethod<ReturnType (ClassType::*)(Args...) const noexcept>'
        struct RegisterClassMethod<ReturnType (ClassType::*)(Args...) const noexcept> {
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1396:8: note: previous definition is here
struct RegisterClassMethod<ReturnType (ClassType::*)(Args...) const noexcept>
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<filament::Renderer *(filament::Engine::*)() noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:485:6: note: in instantiation of function template specialization 'emscripten::class_<filament::Engine>::function<emscripten::internal::DeduceArgumentsTag, filament::Renderer *(filament::Engine::*)() noexcept, emscripten::allow_raw_pointers>' requested here
    .function("createRenderer", &Engine::createRenderer, allow_raw_pointers())
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<filament::View *(filament::Engine::*)() noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:494:6: note: in instantiation of function template specialization 'emscripten::class_<filament::Engine>::function<emscripten::internal::DeduceArgumentsTag, filament::View *(filament::Engine::*)() noexcept, emscripten::allow_raw_pointers>' requested here
    .function("createView", &Engine::createView, allow_raw_pointers())
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<filament::Scene *(filament::Engine::*)() noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:503:6: note: in instantiation of function template specialization 'emscripten::class_<filament::Engine>::function<emscripten::internal::DeduceArgumentsTag, filament::Scene *(filament::Engine::*)() noexcept, emscripten::allow_raw_pointers>' requested here
    .function("createScene", &Engine::createScene, allow_raw_pointers())
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<filament::Camera *(filament::Engine::*)(utils::Entity) noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:517:6: note: in instantiation of function template specialization 'emscripten::class_<filament::Engine>::function<emscripten::internal::DeduceArgumentsTag, filament::Camera *(filament::Engine::*)(utils::Entity) noexcept, emscripten::allow_raw_pointers>' requested here
    .function("getCameraComponent", &Engine::getCameraComponent, allow_raw_pointers())
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<void (filament::View::*)(filament::Camera *) noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:625:6: note: in instantiation of function template specialization 'emscripten::class_<filament::View>::function<emscripten::internal::DeduceArgumentsTag, void (filament::View::*)(filament::Camera *) noexcept, emscripten::allow_raw_pointers>' requested here
    .function("setCamera", &View::setCamera, allow_raw_pointers())
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<void (filament::View::*)(filament::ColorGrading *) noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:626:6: note: in instantiation of function template specialization 'emscripten::class_<filament::View>::function<emscripten::internal::DeduceArgumentsTag, void (filament::View::*)(filament::ColorGrading *) noexcept, emscripten::allow_raw_pointers>' requested here
    .function("setColorGrading", &View::setColorGrading, allow_raw_pointers())
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<void (filament::View::*)(filament::BlendMode) noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:627:6: note: in instantiation of function template specialization 'emscripten::class_<filament::View>::function<emscripten::internal::DeduceArgumentsTag, void (filament::View::*)(filament::BlendMode) noexcept>' requested here
    .function("setBlendMode", &View::setBlendMode)
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<filament::BlendMode (filament::View::*)() const noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:628:6: note: in instantiation of function template specialization 'emscripten::class_<filament::View>::function<emscripten::internal::DeduceArgumentsTag, filament::BlendMode (filament::View::*)() const noexcept>' requested here
    .function("getBlendMode", &View::getBlendMode)
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<void (filament::View::*)(const filament::Viewport &) noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:629:6: note: in instantiation of function template specialization 'emscripten::class_<filament::View>::function<emscripten::internal::DeduceArgumentsTag, void (filament::View::*)(const filament::Viewport &) noexcept>' requested here
    .function("setViewport", &View::setViewport)
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<const filament::Viewport &(filament::View::*)() const noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:630:6: note: in instantiation of function template specialization 'emscripten::class_<filament::View>::function<emscripten::internal::DeduceArgumentsTag, const filament::Viewport &(filament::View::*)() const noexcept>' requested here
    .function("getViewport", &View::getViewport)
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<void (filament::View::*)(unsigned char, unsigned char) noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:631:6: note: in instantiation of function template specialization 'emscripten::class_<filament::View>::function<emscripten::internal::DeduceArgumentsTag, void (filament::View::*)(unsigned char, unsigned char) noexcept>' requested here
    .function("setVisibleLayers", &View::setVisibleLayers)
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<void (filament::View::*)(bool) noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:632:6: note: in instantiation of function template specialization 'emscripten::class_<filament::View>::function<emscripten::internal::DeduceArgumentsTag, void (filament::View::*)(bool) noexcept>' requested here
    .function("setPostProcessingEnabled", &View::setPostProcessingEnabled)
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<void (filament::View::*)(const filament::AmbientOcclusionOptions &) noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:633:6: note: in instantiation of function template specialization 'emscripten::class_<filament::View>::function<emscripten::internal::DeduceArgumentsTag, void (filament::View::*)(const filament::AmbientOcclusionOptions &) noexcept>' requested here
    .function("_setAmbientOcclusionOptions", &View::setAmbientOcclusionOptions)
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<void (filament::View::*)(filament::DepthOfFieldOptions) noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:634:6: note: in instantiation of function template specialization 'emscripten::class_<filament::View>::function<emscripten::internal::DeduceArgumentsTag, void (filament::View::*)(filament::DepthOfFieldOptions) noexcept>' requested here
    .function("_setDepthOfFieldOptions", &View::setDepthOfFieldOptions)
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<void (filament::View::*)(filament::MultiSampleAntiAliasingOptions) noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:635:6: note: in instantiation of function template specialization 'emscripten::class_<filament::View>::function<emscripten::internal::DeduceArgumentsTag, void (filament::View::*)(filament::MultiSampleAntiAliasingOptions) noexcept>' requested here
    .function("_setMultiSampleAntiAliasingOptions", &View::setMultiSampleAntiAliasingOptions)
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<void (filament::View::*)(filament::TemporalAntiAliasingOptions) noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:636:6: note: in instantiation of function template specialization 'emscripten::class_<filament::View>::function<emscripten::internal::DeduceArgumentsTag, void (filament::View::*)(filament::TemporalAntiAliasingOptions) noexcept>' requested here
    .function("_setTemporalAntiAliasingOptions", &View::setTemporalAntiAliasingOptions)
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1630:9: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<void (filament::View::*)(filament::ScreenSpaceReflectionsOptions) noexcept>'
        invoker::template invoke<ClassType, Policies...>(methodName, callable);
        ^
/Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp:637:6: note: in instantiation of function template specialization 'emscripten::class_<filament::View>::function<emscripten::internal::DeduceArgumentsTag, void (filament::View::*)(filament::ScreenSpaceReflectionsOptions) noexcept>' requested here
    .function("_setScreenSpaceReflectionsOptions", &View::setScreenSpaceReflectionsOptions)
     ^
/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot/include/emscripten/bind.h:1344:8: note: template is declared here
struct RegisterClassMethod;
       ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
em++: error: '/Users/wangjunjie/Documents/Github/emsdk/upstream/bin/clang++ -target wasm32-unknown-emscripten -DEMSCRIPTEN -D__EMSCRIPTEN_major__=3 -D__EMSCRIPTEN_minor__=1 -D__EMSCRIPTEN_tiny__=15 -fignore-exceptions -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr -Werror=implicit-function-declaration -Xclang -iwithsysroot/include/SDL --sysroot=/Users/wangjunjie/Documents/Github/emsdk/upstream/emscripten/cache/sysroot -Xclang -iwithsysroot/include/compat -DFILAMENT_IBL_LITE=1 -DFILAMENT_SUPPORTS_OPENGL -DGLTFIO_DRACO_SUPPORTED=1 -I/Users/wangjunjie/Documents/filament/filament/include -I/Users/wangjunjie/Documents/filament/filament/backend/include -I/Users/wangjunjie/Documents/filament/libs/math/include -I/Users/wangjunjie/Documents/filament/libs/utils/include -I/Users/wangjunjie/Documents/filament/third_party/robin-map/tnt/.. -I/Users/wangjunjie/Documents/filament/libs/filaflat/include -I/Users/wangjunjie/Documents/filament/libs/filabridge/include -I/Users/wangjunjie/Documents/filament/libs/ibl/include -I/Users/wangjunjie/Documents/filament/libs/ktxreader/include -I/Users/wangjunjie/Documents/filament/libs/image/include -I/Users/wangjunjie/Documents/filament/third_party/basisu/tnt/../transcoder -I/Users/wangjunjie/Documents/filament/third_party/basisu/tnt/../zstd -I/Users/wangjunjie/Documents/filament/libs/filameshio/include -I/Users/wangjunjie/Documents/filament/out/cmake-webgl-release/libs/gltfio -I/Users/wangjunjie/Documents/filament/libs/gltfio/include -I/Users/wangjunjie/Documents/filament/third_party/cgltf/tnt/.. -I/Users/wangjunjie/Documents/filament/third_party/stb/tnt/.. -I/Users/wangjunjie/Documents/filament/libs/geometry/include -I/Users/wangjunjie/Documents/filament/third_party/hat-trie/tnt/.. -I/Users/wangjunjie/Documents/filament/libs/uberz/include -I/Users/wangjunjie/Documents/filament/out/cmake-webgl-release/third_party/draco/tnt -I/Users/wangjunjie/Documents/filament/third_party/draco/tnt/../src -I/Users/wangjunjie/Documents/filament/libs/viewer/include -I/Users/wangjunjie/Documents/filament/third_party/imgui/tnt/.. -I/Users/wangjunjie/Documents/filament/libs/filagui/include -I/Users/wangjunjie/Documents/filament/third_party/jsmn/tnt/.. -I/Users/wangjunjie/Documents/filament/third_party/civetweb/tnt/../include -std=c++17 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -Wno-deprecated-declarations -fcolor-diagnostics -fvisibility=hidden -DEMSCRIPTEN_HAS_UNBOUND_TYPE_NAMES=0 -O3 -DNDEBUG -fomit-frame-pointer -ffunction-sections -fdata-sections -fno-exceptions -fno-unwind-tables -fno-asynchronous-unwind-tables -fno-rtti -MD -MT web/filament-js/CMakeFiles/filament-js.dir/jsbindings.cpp.o -MF web/filament-js/CMakeFiles/filament-js.dir/jsbindings.cpp.o.d -c /Users/wangjunjie/Documents/filament/web/filament-js/jsbindings.cpp -o web/filament-js/CMakeFiles/filament-js.dir/jsbindings.cpp.o' failed (returned 1)

```

**Desktop (please complete the following information):**
 - OS: [MAC]


**Additional context**
i install the newest version of emsdk from github

--- a/BUILDING.md
+++ b/BUILDING.md
@@ -352,7 +352,7 @@ same version that our continuous builds use.
 
 ```
 cd <your chosen parent folder for the emscripten SDK>
-curl -L https://github.com/emscripten-core/emsdk/archive/refs/tags/3.1.9.zip > emsdk.zip
+curl -L https://github.com/emscripten-core/emsdk/archive/refs/tags/3.1.15.zip > emsdk.zip
 unzip emsdk.zip ; mv emsdk-* emsdk ; cd emsdk
 python ./emsdk.py install latest
 python ./emsdk.py activate latest

--- a/RELEASE_NOTES.md
+++ b/RELEASE_NOTES.md
@@ -5,6 +5,8 @@ A new header is inserted each time a *tag* is created.
 
 ## main branch
 
+- WebGL: upgraded the JS bindings to work with emsdk 3.1.15
+
 ## v1.25.4
 
 - backend: streamline texture upload APIs [⚠️ **API Change**]

--- a/build/web/ci-common.sh
+++ b/build/web/ci-common.sh
@@ -17,7 +17,7 @@ export PATH="$PWD:$PATH"
 # npm install -g typescript
 
 # Install emscripten.
-curl -L https://github.com/emscripten-core/emsdk/archive/refs/tags/3.1.9.zip > emsdk.zip
+curl -L https://github.com/emscripten-core/emsdk/archive/refs/tags/3.1.15.zip > emsdk.zip
 unzip emsdk.zip ; mv emsdk-* emsdk ; cd emsdk
 ./emsdk install latest
 ./emsdk activate latest

--- a/web/filament-js/jsbindings.cpp
+++ b/web/filament-js/jsbindings.cpp
@@ -145,53 +145,6 @@ namespace emscripten {
         BIND(utils::EntityManager)
         BIND(VertexBuffer)
         BIND(View)
-
-        // embind is missing a template definition for "noexcept" methods, so
-        // we're supplying it ourselves while waiting for the upstream fix.
-        template<typename ClassType, typename ReturnType, typename... Args>
-        struct RegisterClassMethod<ReturnType (ClassType::*)(Args...) noexcept> {
-
-            template <typename CT, typename... Policies>
-            static void invoke(const char* methodName,
-                               ReturnType (ClassType::*memberFunction)(Args...) noexcept)  {
-                auto invoker = &MethodInvoker<decltype(memberFunction), ReturnType, ClassType*, Args...>::invoke;
-
-                typename WithPolicies<Policies...>::template ArgTypeList<ReturnType, AllowedRawPointer<ClassType>, Args...> args;
-                _embind_register_class_function(
-                    TypeID<ClassType>::get(),
-                    methodName,
-                    args.getCount(),
-                    args.getTypes(),
-                    getSignature(invoker),
-                    reinterpret_cast<GenericFunction>(invoker),
-                    getContext(memberFunction),
-                    isPureVirtual<Policies...>::value);
-            }
-        };
-
-        // embind is missing a template definition for "const noexcept" methods, so
-        // we're supplying it ourselves while waiting for the upstream fix.
-        template<typename ClassType, typename ReturnType, typename... Args>
-        struct RegisterClassMethod<ReturnType (ClassType::*)(Args...) const noexcept> {
-
-            template <typename CT, typename... Policies>
-            static void invoke(const char* methodName,
-                               ReturnType (ClassType::*memberFunction)(Args...) const noexcept)  {
-                auto invoker = &MethodInvoker<decltype(memberFunction), ReturnType, const ClassType*, Args...>::invoke;
-
-                typename WithPolicies<Policies...>::template ArgTypeList<ReturnType, AllowedRawPointer<const ClassType>, Args...> args;
-                _embind_register_class_function(
-                    TypeID<ClassType>::get(),
-                    methodName,
-                    args.getCount(),
-                    args.getTypes(),
-                    getSignature(invoker),
-                    reinterpret_cast<GenericFunction>(invoker),
-                    getContext(memberFunction),
-                    isPureVirtual<Policies...>::value);
-            }
-        };
-
     }
 }
 #undef BIND

